api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+
+	public void setConditionDAO(ConditionDAO conditionDAO) {
+		this.conditionDAO = conditionDAO;
+	}
+
+	public ConditionDAO getConditionDAO() {
+		return conditionDAO;
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	@Transactional
+	public Condition saveCondition(Condition condition) throws APIException {
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Gets a condition based on the uuid
+	 *
+	 * @param uuid - uuid of the condition to be returned
+	 * @return the condition
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getConditionByUuid(String uuid) {
+		return conditionDAO.getConditionByUuid(uuid);
+	}
+
+	/**
+	 * Gets a condition by id
+	 *
+	 * @param conditionId - the id of the Condition to retrieve
+	 * @return the Condition with the given id, or null if none exists
+	 * @throws APIException
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getCondition(Integer conditionId) throws APIException {
+		return conditionDAO.getCondition(conditionId);
+	}
+
+	/**
+	 * Gets a patient's active conditions
+	 *
+	 * @param patient - the patient to retrieve conditions for
+	 * @return a list of the patient's active conditions
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getActiveConditions(Patient patient) {
+		return conditionDAO.getActiveConditions(patient);
+	}
+
+	/**
+	 * Voids a condition
+	 *
+	 * @param condition  - the condition to be voided
+	 * @param voidReason - the reason for voiding the condition
+	 */
+	@Override
+	@Transactional
+	public Condition voidCondition(Condition condition, String voidReason) {
+		condition.setVoided(true);
+		condition.setVoidedBy(Context.getAuthenticatedUser());
+		condition.setVoidReason(voidReason);
+		if(condition.getDateVoided() == null){
+			condition.setDateVoided(new Date());
+		}
+		Context.getConditionService().saveCondition(condition);
+		return condition;
+	}
+
+	/**
+	 * Revive a condition
+	 *
+	 * @param condition Condition to unvoid
+	 * @throws APIException
+	 * @should unset voided bit on given condition
+	 */
+	@Override
+	@Transactional
+	public Condition unvoidCondition(Condition condition) throws APIException {
+		condition.setVoided(false);
+		condition.setVoidedBy(null);
+		condition.setDateVoided(null);
+		condition.setVoidReason(null);
+		Context.getConditionService().saveCondition(condition);
+		return condition;
+	}
+
+	/**
+	 * Completely remove a condition from the database. This should typically not be called
+	 * because we don't want to ever lose data. The data really <i>should</i> be voided and then it
+	 * is not seen in interface any longer (see #voidCondition(Condition) for that one) If other things link to
+	 * this condition, an error will be thrown.
+	 *
+	 * @param condition
+	 * @throws APIException
+	 * @should delete the given condition from the database
+	 */
+	@Override
+	@Transactional(readOnly = false)
```

Comment:
You will not need this after putting it at the class level

Reply:
This is noted